core/src/main/scala/io/qbeast/core/model/MetadataManager.scala

```diff
@@ -8,8 +8,10 @@ import io.qbeast.IISeq
  *   type of data schema
  * @tparam FileDescriptor
  *   type of file descriptor
+ * @tparam QbeastOptions
+ *   type of the Qbeast options
  */
-trait MetadataManager[DataSchema, FileDescriptor] {
+trait MetadataManager[DataSchema, FileDescriptor, QbeastOptions] {
```

Comment:
Should QbeastOptions be an interface in `core` instead of a type?

Reply:
I just follow the approach chosen by @cugni. IMHO the core as a separate project with generic interfaces is overdesign here.

Reply:
QbeastOptions does not contain any specific Spark class, so it could be put directly in the core, with no need for generics. Indeed, DataSchema and FileDescriptor are generics because they are a specific class of Spark. The argument was that in the future, we might want to write in Qbeast without using Spark, which is something that now, with the Delta Kernels, is also possible and it might be useful. IMHO, the QbeastOptions relates only to Delta, nor to Spark, so they should be in core.